### PR TITLE
HandledException

### PR DESCRIPTION
**Pull request by AayushSaini101**
Added Exception Handling for file not found Exception

### CODE REVIEW

**Comment by rnveach** [comment]
@AayushSaini101 Assuming this is connected to #10749, please tie it back to that issue. You must reference issue in first post and commit. Run tests locally and they will guide you on how to make the commit. You should always run local tests before pushing to the server.

You do not need to close the PR to fix it up.

Also look at the issue on how it was brought forward that this should be fixed at https://github.com/checkstyle/checkstyle/issues/10749#issue-981933276 .
> The Util should be changed so that those 3 folders are taken as required inputs and not hardcoded in.

File: src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java [hunk 1]
```diff
@@ -80,19 +81,23 @@ private static void dumpMetadata(Checker checker, String path) throws Checkstyle
...
-                            .filter(file -> {
-                                return file.getName().endsWith("SuppressWarningsHolder.java")
-                                        || file.getName().endsWith("Check.java")
-                                        || file.getName().endsWith("Filter.java");
-                            })
-                            .collect(Collectors.toList()));
+            try {
+                final List<String> moduleFolders = Arrays.asList("checks", "filters", "filefilters");
+                for (String folder : moduleFolders) {
+                    try (Stream<Path> files = Files.walk(Paths.get(path
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java:87]: Please replace `Paths.get(path + "/" + folder)` => `Paths.get(path, folder)`

The "/" symbol is FS-specific.

File: src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java [hunk 2]
```diff
@@ -80,19 +81,23 @@ private static void dumpMetadata(Checker checker, String path) throws Checkstyle
...
+                                        .filter(file -> {
+                                            return file.getName().endsWith("SuppressWarningsHolder.java")
+                                                    || file.getName().endsWith("Check.java")
+                                                    || file.getName().endsWith("Filter.java");
+                                        })
+                                        .collect(Collectors.toList()));
+                    }
                 }
+            }catch (FileNotFoundException e){
+                System.out.println("Folders not present");
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java:100]: Which folder? This information is present in the exception. Why not share it with the user?

**Review by pbludov — CHANGES_REQUESTED** [review]
Please make the CI happy

File: src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java (hunk 2 above)
Comment (AayushSaini101) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java:100]: @pbludov  Thanks sir I will check again

Comment (AayushSaini101) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java:100]: Sir how to share this with the user

Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java:100]: > Sir how to share this with the user

Like this (for example):
```java
  System.out.println("Folder not present: " + exception.getFile());
```
Note that `Files.walk` throws `NoSuchFileException`, not `FileNotFoundException`.

**Review by AayushSaini101 — COMMENTED** [review]
Resolve #10749 
Added NoSuchFileException handling

**Review by AayushSaini101 — COMMENTED** [review]
Resolve #10749 
Added NoSuchFileException Handling

**Review by AayushSaini101 — COMMENTED** [review]
Resolve Issue #10749 
Added NoSuchFileException Handling

**Comment by AayushSaini101** [comment]
@pbludov Sir can i make new PR request

**Comment by pbludov** [comment]
> @pbludov Sir can i make new PR request

have you read https://checkstyle.org/beginning_development.html and https://checkstyle.org/contributing.html ?

**Comment by nmancus1** [comment]
Closed as abandoned.

**Pull request closed without merging**